pkg/network/usm/kafka_monitor_test.go

```diff
@@ -131,10 +133,7 @@ func (s *KafkaProtocolParsingSuite) testKafkaProtocolParsing(t *testing.T, tls b
 	serverHost := "127.0.0.1"
 
 	kafkaTeardown := func(t *testing.T, ctx testContext) {
-		if _, ok := ctx.extras["client"]; !ok {
-			return
-		}
-		if client, ok := ctx.extras["client"].(*kafka.Client); ok {
+		for _, client := range ctx.clients {
 			defer client.Client.Close()
```

Comment:
no need to use defer here

Reply:
Right, I wasn't sure why it was done that way, and I left that line alone since I didn't need to touch it for this fix. I'll send a separate PR to change that.